### PR TITLE
[Feature][Connector]Spark connector for excel Source

### PR DESCRIPTION
**Pull request by tmljob**
## Purpose of this pull request

Spark connector for excel Source

## Check list

* [x] Code changed are covered with tests, or it does not need tests for reason:
* [x] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [x] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs

### CODE REVIEW

**Comment by wuchunfu** [comment]
@tmljob Please resolve the conflicting files, thanks.

File: docs/en/connector/source/Excel.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+file path, starting with file://
+
+### options.useHeader[boolean]
+
+Whether to use the first row of Excel as the header
+
+### options.*
+
+For details, refer to the parameters of the third-party library [spark-excel](https://github.com/crealytics/spark-excel).
```
Comment (Hisoka-X) [inline docs/en/connector/source/Excel.md]: After read this document, I can't know how to set options.*, I think move there option to here is better.

**Comment by Hisoka-X** [comment]
Can this plugin support custom header( or custom fields) in config now?

File: docs/en/connector/source/Excel.md (hunk 1 above)
Comment (tmljob) [reply docs/en/connector/source/Excel.md]: > Can this plugin support custom header( or custom fields) in config now?

1、You can set whether to use the first line as the header, but it cannot be customized
2、You can set the area range for reading data through "DataAddress"

**Comment by tmljob** [comment]
@BenJFan @wuchunfu @CalvinKirs 
Is there anything else that needs to be modified in my PR? Why can't it be merged into dev?

**Comment by CalvinKirs** [comment]
> @BenJFan @wuchunfu @CalvinKirs Is there anything else that needs to be modified in my PR? Why can't it be merged into dev?

can you resolve CI fail first?

**Comment by tmljob** [comment]
> > @BenJFan @wuchunfu @CalvinKirs Is there anything else that needs to be modified in my PR? Why can't it be merged into dev?
> 
> can you resolve CI fail first?

I don't understand why the build fails under ubuntu, and no clear error message is provided.

**Comment by tmljob** [comment]
> > > @BenJFan @wuchunfu @CalvinKirs Is there anything else that needs to be modified in my PR? Why can't it be merged into dev?
> > 
> > 
> > can you resolve CI fail first?
> 
> I don't understand why the build fails under ubuntu, and no clear error message is provided.

@CalvinKirs 
Is CI OK now, only one task is canceled by timeout.

**Comment by CalvinKirs** [comment]
> > > > @BenJFan @wuchunfu @CalvinKirs Is there anything else that needs to be modified in my PR? Why can't it be merged into dev?
> > > 
> > > 
> > > can you resolve CI fail first?
> > 
> > 
> > I don't understand why the build fails under ubuntu, and no clear error message is provided.
> 
> @CalvinKirs Is CI OK now, only one task is canceled by timeout.

don't worry~

File: seatunnel-dist/release-docs/LICENSE [hunk 2]
```diff
@@ -896,6 +896,7 @@ The text of each license is the standard Apache 2.0 license.
      (The Apache Software License, Version 2.0) transport-netty4 (org.elasticsearch.plugin:transport-netty4-client:7.5.1 - https://github.com/elastic/elasticsearch)
      (The Apache Software License, Version 2.0) x-content (org.elasticsearch:elasticsearch-x-content:6.3.1 - https://github.com/elastic/elasticsearch)
      (The Apache Software License, Version 2.0) zookeeper (org.apache.zookeeper:zookeeper:3.4.10 - no url defined)
+     (The Apache Software License, Version 2.0) spark-excel (com.crealytics:spark-excel_2.11:0.11.1 - https://github.com/crealytics/spark-excel)
```
Comment (CalvinKirs) [inline seatunnel-dist/release-docs/LICENSE:899]: Other licenses also need to be added here

File: tools/dependencies/known-dependencies.txt [hunk 3]
```diff
@@ -707,4 +707,17 @@ zookeeper-3.4.6.jar
 zookeeper-3.5.9.jar
 zookeeper-jute-3.5.9.jar
 zstd-jni-1.3.3-1.jar
-zstd-jni-1.4.3-1.jar
\ No newline at end of file
+zstd-jni-1.4.3-1.jar
```
Comment (Hisoka-X) [inline tools/dependencies/known-dependencies.txt]: The corresponding dependencies should be put into the corresponding lines, according to lexicographical order

File: seatunnel-dist/release-docs/LICENSE (hunk 2 above)
Comment (tmljob) [reply seatunnel-dist/release-docs/LICENSE:899]: I introduced the spark-excel tripartite package. In addition to listing the license of spark-excel, I also need to list the licesne of the library it depends on, right?

Comment (Hisoka-X) [reply seatunnel-dist/release-docs/LICENSE:899]: > I introduced the spark-excel tripartite package. In addition to listing the license of spark-excel, I also need to list the licesne of the library it depends on, right?

Right, just add license which that you add jar name in know-dependencies.txt

Comment (Hisoka-X) [reply seatunnel-dist/release-docs/LICENSE:899]: Seem like you should add other license in this file. like 
```
poi-4.0.0.jar
poi-ooxml-4.0.0.jar
poi-ooxml-schemas-4.0.0.jar
```
They are new jar you add on known-dependcies.txt, so you should add license link on this file, you can copy example from already exist license.

**Comment by Hisoka-X** [comment]
@CalvinKirs Hi, please check if the license is legal

**Comment by EricJoy2048** [comment]
This pr is closed because it has not responded for too long. If necessary, please reopen it

**Pull request closed without merging**